src/api/admin.ts

```diff
@@ -32,6 +32,17 @@ export type ListCellIdsResponse = Array<CellId>
 export type ListActiveAppsRequest = void
 export type ListActiveAppsResponse = Array<InstalledAppId>
 
+export type AgentInfoSigned = {
+    agent: AgentPubKey,
```

Comment:
Ok this is wrong. It's not an `AgentPubKey`.
The `AgentInfoSigned` type should never be used at this layer.
There's no need to deserialize the type at all, just take the array of bytes from the websocket and pass it to add agent info

Reply:
Hmmm.  To not deserialize the data from a message requires changing the plumbing quite a bit.  The code is generalized to deserialize every message type.  But even if this is "wrong" as an interpretation of the data when deserialized, I think it is still working because in the test when it is reserialized, and passed into AddAgentInfo, and then pulled back out and compared with the original data it all matches.  AgentPubKey happens to be just a Buffer. So even though we don't "need" to serialize and reserialize, that doesn't seem to be a problem.  Just incase I changed the type to "any" and added the deepequal test.